src/main/java/data/task/TaskList.java

```diff
@@ -0,0 +1,296 @@
+package data.task;
+
+import data.exception.DukeException;
+import parser.Parser.Command;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class TaskList {
+    /** The ArrayList for storing all the tasks. */
+    private ArrayList<Task> tasks;
+    /** The file for writing/reading the task to/from. */
+    private File file;
+
+    /**
+     * Constructs a new TaskList.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<Task>();
+    }
+
+    /**
+     * Constructs a new TaskList based on the specified file.
+     *
+     * @param file The file to write to or read from.
+     */
+    public TaskList(File file) {
+        this.tasks = new ArrayList<Task>();
+        this.file = file;
+    }
+
+    /**
+     * Reads the contents of the file and stores it into the ArrayList.
+     */
+    public void open() {
+        this.tasks = new ArrayList<Task>();
+        try {
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String storedTask = scanner.nextLine();
+                String[] taskArray = storedTask.split(",");
+                Task task;
+                if (taskArray[0].startsWith("Todo")) {
+                    task = new ToDoTask(taskArray[2]);
+                } else if (taskArray[0].startsWith("Deadline")) {
+                    task = new DeadlineTask(taskArray[2], taskArray[3]);
+                } else {
+                    task = new EventTask(taskArray[2], taskArray[3], taskArray[4]);
+                }
+                if ((taskArray[1]).equals("1")) {
+                    task.setDone();
+                }
+                this.tasks.add(task);
+            }
+            scanner.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("An error occurred when trying to find the file.");
+            e.getMessage();
+            e.printStackTrace();
+        } catch (DukeException e) {
+            System.out.println(e);
+        }
+    }
+
+    /**
+     * Calls a specific method with the specified input based on the specified command.
+     *
+     * @param command The command to be run.
+     * @param input The input string for the command.
+     * @throws DukeException If an EventTask is instantiated with invalid dates.
+     */
+    public void executeCommand(Command command, String input) throws DukeException {
```

Comment:
I like that you delegated the logic of each command to their own function. It reduced the overall length of the methods